lib/MonadicRewrite.thy

```diff
@@ -368,19 +408,39 @@ lemma monadic_rewrite_if:
                           (If P a b) (If P c d)"
   by (cases P, simp_all)
 
-lemma monadic_rewrite_if_rhs:
+lemma monadic_rewrite_if_r:
   "\<lbrakk> P \<Longrightarrow> monadic_rewrite F E Q a b; \<not> P \<Longrightarrow> monadic_rewrite F E R a c \<rbrakk>
    \<Longrightarrow> monadic_rewrite F E (\<lambda>s. (P \<longrightarrow> Q s) \<and> (\<not> P \<longrightarrow> R s)) a (If P b c)"
   by (cases P, simp_all)
 
+lemma monadic_rewrite_if_r_True:
+  "\<lbrakk> P \<Longrightarrow> monadic_rewrite F E Q a b \<rbrakk>
+   \<Longrightarrow> monadic_rewrite F E ((\<lambda>_. P) and Q) a (If P b c)"
```

Comment:
I'm not sure... it seems like when we talk about `If` from the perspective of a construct (i.e. there's no further detail on the contents of the predicate and branches) we stick to `If` because it's shorter. I could change it, but not sure it helps much.